generator/internal/genclient/translator/openapi/openapi.go

```diff
@@ -132,66 +140,145 @@ func makeMessageFields(messageName string, message *base.Schema) ([]*genclient.F
 	return fields, nil
 }
 
-func makeField(messageName, name string, field *base.Schema) (*genclient.Field, error) {
+func makeField(messageName, name string, optional bool, field *base.Schema) (*genclient.Field, error) {
+	if len(field.AllOf) != 0 {
+		return makeAllOfField(messageName, name, field)
+	}
 	if len(field.Type) == 0 {
 		return nil, fmt.Errorf("missing field type for field %s.%s", messageName, name)
 	}
 	type_name := field.Type[0]
-	optional := true
-	for _, r := range field.Required {
-		if name == r {
-			optional = false
-			break
-		}
-	}
 	switch type_name {
-	case "string":
+	case "boolean":
 		return &genclient.Field{
 			Name:          name,
 			Documentation: field.Description,
-			Typez:         genclient.STRING_TYPE,
+			Typez:         genclient.BOOL_TYPE,
 			Optional:      optional,
 		}, nil
+	case "string":
+		return makeStringType(messageName, name, field.Format, optional, field)
 	case "integer":
-		{
-			typez, err := makeIntegerType(field.Format)
-			if err != nil {
-				return nil, err
-			}
-			return &genclient.Field{
-				Name:          name,
-				Documentation: field.Description,
-				Typez:         typez,
-				Optional:      optional,
-			}, nil
-		}
+		return makeIntegerField(messageName, name, field.Format, optional, field)
 	case "object":
 		return &genclient.Field{
 			Name:          name,
 			Documentation: field.Description,
 			Typez:         genclient.MESSAGE_TYPE,
 			Optional:      true,
 		}, nil
-
 	case "array":
 		return makeArrayField(messageName, name, field)
 	default:
 		return nil, fmt.Errorf("unknown type for field %q", name)
 	}
 }
 
-func makeIntegerType(format string) (genclient.Typez, error) {
+func makeAllOfField(messageName, name string, field *base.Schema) (*genclient.Field, error) {
+	for _, proxy := range field.AllOf {
+		typezID := strings.TrimPrefix(proxy.GetReference(), "#/components/schemas/")
+		return &genclient.Field{
+			Name:          name,
+			Documentation: field.Description,
+			Typez:         genclient.MESSAGE_TYPE,
+			TypezID:       typezID,
+			Optional:      true,
+		}, nil
+	}
+	return nil, fmt.Errorf("cannot build any AllOf schema for field %s.%s", messageName, name)
+}
+
+func makeStringType(messageName, name, format string, optional bool, field *base.Schema) (*genclient.Field, error) {
+	switch format {
+	case "":
```

Comment:
should this be case "string"?

Reply:
No.  A normal string has no `"format"` field. Integers as strings have  `{"type": "string", "format": "int64"}`.

I sure don't like how OpenAPI represents primitive types 🤷